NineChronicles.Headless/GraphTypes/StandaloneQuery.cs

```diff
@@ -72,6 +74,87 @@ public StandaloneQuery(StandaloneContext standaloneContext, IConfiguration confi
                 }
             );
 
+            Field<ListGraphType<RootStateDiffType>>(
+                name: "diffs",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<LongGraphType>>
+                    {
+                        Name = "baseIndex",
+                        Description = "The index of the block used to fetch state from chain."
+                    },
+                    new QueryArgument<NonNullGraphType<LongGraphType>>
+                    {
+                        Name = "changedIndex",
+                        Description = "The index of the block used to fetch state from chain."
+                    }
+                ),
+                resolve: context =>
+                {
+                    if (!(standaloneContext.BlockChain is BlockChain blockChain))
+                    {
+                        throw new ExecutionError(
+                            $"{nameof(StandaloneContext)}.{nameof(StandaloneContext.BlockChain)} was not set yet!"
+                        );
+                    }
+
+                    var baseIndex = context.GetArgument<long>("baseIndex");
+                    var changedIndex = context.GetArgument<long>("changedIndex");
+
+                    var blockInterval = changedIndex - baseIndex;
+
+                    if (blockInterval >= 10 && blockInterval > 0)
+                    {
+                        throw new ExecutionError(
+                            "Interval between baseIndex and changedIndex should under 10"
+                        );
+                    }
```

Comment:
Did you intend like below?
```suggestion
                    var blockInterval = changedIndex - baseIndex;
                    if (blockInterval < 0 || blockInterval >= 10)
                    {
                        throw new ExecutionError(
                            "Interval between baseIndex and changedIndex should be greater than or equal to 0 and less than 10."
                        );
                    }
```

Reply:
Thanks, done!

Reply:
Why not increase the minimum value of `blockInterval` from `0` to `1`?
I thought if `blockInterval` was `0`, there would be no diff.

Reply:
Since there are no limits for `changedIndex`, `baseIndex`, the `blockInterval` can be negative. I think it is normal case.

Reply:
Oh, and it also provides a lookup for reverse changes, right?

Reply:
Yes, you're right. It is just trie diff.

Reply:
If so, I'll have to write it differently, not as suggested in this thread.
I'll also need to revise the description of the query arguments.🙂

Reply:
Even if 0 is allowed, the GraphQL server ultimately does nothing and just returns an empty list.
Therefore, I thought it should not be allowed in the validation phase.

Reply:
> Even if 0 is allowed, the GraphQL server ultimately does nothing and just returns an empty list.
Therefore, I thought it should not be allowed in the validation phase.

If so, my suggestion above would be valid.

Reply:
> Why not increase the minimum value of `blockInterval` from `0` to `1`? I thought if `blockInterval` was `0`, there would be no diff.

Sorry I misunderstood, However changed the rule, please check it

Reply:
> Sorry I misunderstood, However changed the rule, please check it

Oh, never mind.😉